src/commands/lambda/__tests__/functions/instrument.test.ts

```diff
@@ -458,7 +458,181 @@ describe('instrument', () => {
         '`apiKeySecretArn` is not supported for Node runtimes when using Synchronous Metrics. Use either `apiKey` or `apiKmsKey`.'
       )
     })
+
+    describe('test the new universal instrumentation workflow for Java and .Net', () => {
+      const region = 'us-east-1'
+      const config = {
+        FunctionArn: 'arn:aws:lambda:us-east-1:123456789012:function:lambda-hello-world',
+        Handler: 'index.handler',
+        Layers: [],
+        Runtime: 'runtime',
+      }
+      const settings = {
+        flushMetricsToLogs: false,
+        layerAWSAccount: mockAwsAccount,
+        layerVersion: 13,
+        mergeXrayTraces: false,
+        tracingEnabled: false,
+      }
+      const dotnetRuntime = 'dotnet6'
+      const javaRuntime = 'java11'
+
+      describe('test for runtime ${dotnetRuntime}', () => {
+        const dotNetConfig = {...config, Runtime: dotnetRuntime}
+
+        test('should throw error when the extension version and trace version are not compatible', async () => {
+          process.env[CI_KMS_API_KEY_ENV_VAR] = '5678'
+          const badSettings = {...settings, extensionVersion: 24, layerVersion: 3}
+          let error
+          try {
+            await calculateUpdateRequest(dotNetConfig, badSettings, region, dotnetRuntime)
+          } catch (e) {
+            if (e instanceof Error) {
+              error = e
+            }
+          }
+          expect(error?.message).toBe(
+            `For the ${dotnetRuntime} runtime, the dd-trace version 3 is not compatible with the dd-extension version 24`
+          )
+        })
+
+        const baseVariables = {
+          DD_KMS_API_KEY: '5678',
+          DD_MERGE_XRAY_TRACES: 'false',
+          DD_SITE: 'datadoghq.com',
+          DD_TRACE_ENABLED: 'false',
+        }
+        const compatibleTradeAndExtension = {
```

Comment:
I think this is a typo, `compatibleTradeAndExtension` -> `compatibleTracerAndExtension`

Reply:
Good catch